### PR TITLE
tbi(btp-utils): handle missing types and func - Part2

### PR DESCRIPTION
**Pull request by longieirl**
Fix for https://github.com/SAP/open-ux-tools/issues/1633

- additional tests added
- expose existing types
- Add new functionalty to handle displaying destination details, used in the UI

### CODE REVIEW

**Review by zdravko-georgiev — APPROVED** [review]
Code looks good
Excellent test coverage
Changeset OK

File: packages/btp-utils/src/destination.ts [hunk 1]
```diff
@@ -136,9 +160,67 @@ export function isOnPremiseDestination(destination: Destination): boolean {
...
+ * Checks whether the provided destination is configured to point to an S/4 HANA system.
+ *
+ * @param destination destination info
+ * @returns boolean if the destination is configured for an SAP S/4HANA system
+ */
+export function isS4HC(destination: Destination): boolean {
+    return Boolean(
+        destination.WebIDEUsage?.includes(WebIDEUsage.ODATA_ABAP) &&
+            destination.Authentication === Authentication.SAML_ASSERTION &&
+            destination.ProxyType === ProxyType.INTERNET
```
Comment (longieirl) [inline packages/btp-utils/src/destination.ts:224]: Reviewed locally and updated to reflect ProxyType of `Internet`.

**Review by longieirl — COMMENTED** [review]
Minor change based on local discussion.

**Review by cianmSAP — APPROVED** [review]
- Code look good and easy to understand
- 100% coverage
- Changeset ✅
- Did not test locally

**Review by zdravko-georgiev — APPROVED** [review]
re-approving

**Pull request merged**